### PR TITLE
Carousel Pre-Rendering Fixes + Cleanup

### PR DESCRIPTION
**Pull request by sghoweri**
## Jira
http://vjira2:8080/browse/BDS-1982?filter=-1

## Summary
Updates the Carousel component' to significantly speed up the perceived rendering performance + reduce the overall amount of thrashing before vs after the component's JS has booted up.

## Details

image

[Link to before vs after comparison](https://www.webpagetest.org/video/compare.php?tests=200127_JN_e02873d4a44be6f9326b9acc5ecdbeb2-r%3A3-c%3A0-l%3ABefore%2C200127_X1_65e2e6fbc290fb40e265fa10b69bab4d-r%3A1-c%3A0-l%3AAfter%2C&thumbSize=200&ival=100&end=full)

- Updates the slides-per-view demo to use smaller placeholder containers that can fully fit the space available at smaller screen sizes
- Consolidates + simplifies the CSS logic handling the # of slides per view displayed
  - NOTE: while I had originally made this updates with only CSS variable support in mind, it didn't feel right to completely remove support (ie. pre-rendering) in just this one spot vs an across the board change. Because of that, I ended up adding styles for handling this fallback size / space + progressively enhanced that if CSS variables are supported. 
- Eliminates the need to conditionally provide extra space around Carousel slides containing cards (now it always "just works"
- Updates Swiper to the latest version on NPM + switches over to use ES Module imports to help improve JS rendering performance

### API Changes
- The `slide_offset_before` and `slide_offset_after` props are no longer needed since this behavior is now built in!
- The maximum # of slides that can be displayed at once has been capped off at **3** (was previously 6!) given that the primary use case to display more than 3 slides at once (ex. an image gallery view) isn't yet fully baked
  - Furthermore, this change also helps to simplify the overall amount of CSS + JS logic being maintained under the hood

## How to test
- Review carousel demos in Chrome (etc al.) + IE 11 to confirm no major visual regressions + layout issues

### CODE REVIEW

File: packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    min-width: calc(100% - calc(var(--bolt-carousel-space-around) * 2));
+  }
+
+  bolt-carousel[slides-per-view="2"] & {
+    min-width: calc(100% - calc(#{$bolt-carousel-space-around} * 2));
+    min-width: calc(100% - calc(var(--bolt-carousel-space-around) * 2));
+
+    @include bolt-mq($from: 600px) {
+      min-width: calc(calc(100% / 2) - calc(#{$bolt-carousel-space-around} / 2) - calc(#{$bolt-carousel-space-around} * 1));
+      min-width: calc(calc(100% / 2) - calc(var(--bolt-carousel-space-between) / 2) - calc(var(--bolt-carousel-space-around) * 1));
```
Comment (mikemai2awesome) [inline packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss]: Is there a typo here? First line is using `space-around / 2` and second line is using `space-between / 2`.

File: packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+  bolt-carousel:not([slides-per-view="1"]):not([slides-per-view="2"]) &,
+  bolt-carousel[slides-per-view="auto"] &,
+  bolt-carousel[slides-per-view="3"] & {
+    min-width: calc(100% - calc(#{$bolt-carousel-space-around} * 2));
+    min-width: calc(100% - calc(var(--bolt-carousel-space-around) * 2));
+
+    @include bolt-mq($from: 600px) {
+      min-width: calc(calc(100% / 2) - calc(#{$bolt-carousel-space-around} / 2) - calc(#{$bolt-carousel-space-around} * 1));
+      min-width: calc(calc(100% / 2) - calc(var(--bolt-carousel-space-between) / 2) - calc(var(--bolt-carousel-space-around) * 1));
```
Comment (mikemai2awesome) [inline packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss]: Same with this, typo?

File: packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    }
+
+    @include bolt-mq($from: medium) {
+      min-width: calc(calc(100%  / 3) - calc(#{$bolt-carousel-space-around} / 3) -  calc(#{$bolt-carousel-space-around} * 3));
+      min-width: calc(calc(100%  / 3) - calc(var(--bolt-carousel-space-between) / 3) -  calc(var(--bolt-carousel-space-around) * 3));
```
Comment (mikemai2awesome) [inline packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss]: Same with this, typo?

File: packages/components/bolt-carousel/src/_carousel.scss [hunk 4]
```diff
@@ -54,7 +52,12 @@ bolt-carousel {
   }
 
   &--no-overflow {
+    min-width: calc(100% + var(--bolt-carousel-space-around) * 2);
     overflow: hidden;
+    margin-right: calc(var(--bolt-carousel-space-around) * -1);
+    margin-left: calc(var(--bolt-carousel-space-around) * -1);
+    padding-right: var(--bolt-carousel-space-around);
+    padding-left:var(--bolt-carousel-space-around);
```
Comment (mikemai2awesome) [inline packages/components/bolt-carousel/src/_carousel.scss]: Why is this using single declaration and CSS vars only as opposed to the double declarations method elsewhere?

Comment (sghoweri) [reply packages/components/bolt-carousel/src/_carousel.scss]: @mikemai2awesome I'll confirm why (and if this is still needed) but I _think_ these styles only apply to browsers that are rendering the Carousel with an actual <slot> (aka only browsers that support CSS custom properties). 

I'll double-check and report back in this.

File: packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss (hunk 1 above)
Comment (sghoweri) [reply packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss]: @mikemai2awesome this is most likely a typo when I was trying to retrofit IE 11 support. 🤦‍♂️

I've updated all of these Sass vars flagged below!

File: packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss (hunk 3 above)
Comment (sghoweri) [reply packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss]: Updated the Sass. We'll need to double-check that this is still working correctly.

File: packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss (hunk 2 above)
Comment (sghoweri) [reply packages/components/bolt-carousel/src/_carousel-slide.slides-per-view.scss]: Updated the Sass. We'll need to double-check that this is still working correctly.

**Comment by sghoweri** [comment]
> @sghoweri I did a `clean` and `setup` on this branch, started it up and got this:
> 
> image

@danielamorse That's from the Sass variable name change I tossed in yesterday per Mike's comment. This morning I still need to put in my todo comment to further consolidate the slide width vs # of slides logic so I'll get that fixed as well.

**Comment by sghoweri** [comment]
PR was released with v2.17.0

**Pull request merged**